pkg/admission/workspace/admission.go

```diff
@@ -163,27 +165,40 @@ func (o *workspace) Validate(ctx context.Context, a admission.Attributes, _ admi
 			return fmt.Errorf("failed to convert unstructured to ClusterWorkspace: %w", err)
 		}
 
+		if old.Spec.Cluster != cw.Spec.Cluster && !isSystemPrivileged {
+			return admission.NewForbidden(a, errors.New("spec.Cluster can only be changed by systems users"))
+		}
+		if old.Spec.URL != cw.Spec.URL && !isSystemPrivileged {
+			return admission.NewForbidden(a, errors.New("spec.URL can only be changed by systems users"))
+		}
+
 		if errs := validation.ValidateImmutableField(cw.Spec.Type, old.Spec.Type, field.NewPath("spec", "type")); len(errs) > 0 {
 			return admission.NewForbidden(a, errs.ToAggregate())
 		}
 		if old.Spec.Type.Path != cw.Spec.Type.Path || old.Spec.Type.Name != cw.Spec.Type.Name {
 			return admission.NewForbidden(a, errors.New("spec.type is immutable"))
 		}
 
-		if old.Status.Cluster != "" && cw.Status.Cluster == "" {
-			return admission.NewForbidden(a, errors.New("status.cluster cannot be unset"))
+		if old.Spec.Cluster != "" && cw.Spec.Cluster == "" {
+			return admission.NewForbidden(a, errors.New("spec.cluster cannot be unset"))
 		}
 
 		if cw.Status.Phase != corev1alpha1.LogicalClusterPhaseScheduling {
```

Comment:
this is still a pretty hard condition on status. Not sure it leads to problems.

Reply:
👍 I'll keep an eye out for errors in the e2e, although this isn't ideal to check for the Phase. We could just remove these checks instead?